src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1065,10 +1065,7 @@ private int FormatNumeric(int value, Span<byte> destination)
                 // GNU format: store negative numbers in big endian format with leading '0xff' byte.
                 //             store positive numbers in big endian format with leading '0x80' byte.
                 long destinationValue = value;
-                if (value >= 0)
```

Comment:
I'm not sure what comment you mean. The `int` values can always be stored in the 8-byte field using the GNU format. It supports the entire range from `int.MinValue` to `int.MaxValue`.

This code extends the int to a long. Because it is 2-complement, negative values are extended with all 0xff.
For positive we need to set the highest bit. We don't do it conditionally on `value` (to avoid a branch) because also for negative numbers the highest bit is set.